rmf_site_format/src/workcell.rs

```diff
@@ -289,6 +344,110 @@ impl Workcell {
+            .map(|c| {
+                let collision = c.1.bundle;
+                let collision = urdf_rs::Collision {
+                    name: Some(collision.name),
+                    origin: collision.pose.into(),
+                    geometry: collision.geometry.into(),
+                };
+                (collision, c.1.parent)
+            })
+            .collect();
+
+        let links: Result<Vec<_>, WorkcellToUrdfError> = workcell
+            .frames
+            .into_iter()
+            .map(|f| {
+                let frame = f.1.bundle;
+
+                let name = match frame.name {
+                    Some(name) => name.0,
+                    None => format!("frame_{}", f.0),
+                };
+                let visual: Vec<Visual> = visuals_and_parents
+                    .iter()
+                    .filter(|(_, parent)| parent == &f.1.parent)
+                    .map(|(visual, _)| visual.clone())
+                    .collect();
+                let collision: Vec<Collision> = collisions_and_parents
+                    .iter()
+                    .filter(|(_, parent)| parent == &f.1.parent)
+                    .map(|(collision, _)| collision.clone())
+                    .collect();
+
+                let pose: urdf_rs::Pose;
+                if let Anchor::Pose3D(p) = frame.anchor {
+                    pose = p.into();
+                } else {
+                    return Err(WorkcellToUrdfError::InvalidAnchorType(frame.anchor));
+                }
+                let link = urdf_rs::Link {
+                    name,
+                    inertial: urdf_rs::Inertial {
+                        origin: pose,
+                        inertia: {
+                            urdf_rs::Inertia {
+                                ixx: 0.0,
+                                ixy: 0.0,
+                                ixz: 0.0,
+                                iyy: 0.0,
+                                iyz: 0.0,
+                                izz: 0.0,
+                            }
+                        },
+                        mass: urdf_rs::Mass { value: 0.0 },
+                    },
+                    collision,
+                    visual,
+                };
+                Ok(link)
+            })
+            .collect();
+        let links = match links {
+            Ok(links) => links,
+            Err(e) => return Err(e),
+        };
+        let robot = urdf_rs::Robot {
+            name: workcell.properties.name,
+            links,
+            joints: vec![],
+            materials: vec![],
+        };
+        return Ok(robot);
+    }
+
+    pub fn to_urdf_string(&self) -> Result<String, WorkcellToUrdfError> {
+        match self.to_urdf() {
+            Ok(urdf) => match urdf_rs::write_to_string(&urdf) {
+                Ok(string) => Ok(string),
+                Err(e) => Err(WorkcellToUrdfError::WriteToStringError(e)),
+            },
+            Err(e) => Err(e),
+        }
```

Comment:
Similar to the above, you can change the outer match statement to:

```
let urdf = self.to_urdf()?;
[inner match statement]
```

Reply:
Done in c8ef823.